### PR TITLE
[tree] Clarify error message

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

**Review by jalopezg-git — APPROVED** [review]

**Review by jalopezg-git — CHANGES_REQUESTED** [review]
Uh, I was too quick to approve, `tree/dataframe/test/dataframe_snapshot.cxx` has to be changed too :slightly_smiling_face:!

**Review by vepadulano — APPROVED** [review]
Yep, makes sense, thank you!

File: tree/tree/src/TTree.cxx [hunk 1]
```diff
@@ -1249,7 +1249,7 @@ bool CheckReshuffling(TTree &mainTree, TTree &friendTree)
 
    if ((isMainReshuffled || isFriendReshuffled) && !friendHasValidIndex) {
       const auto reshuffledTreeName = isMainReshuffled ? mainTree.GetName() : friendTree.GetName();
-      const auto msg = "Tree '%s' has the kEntriesReshuffled bit set, and cannot be used as friend nor can be added as "
+      const auto msg = "Tree '%s' has the kEntriesReshuffled bit set and cannot have friends nor can be added as "
                        "a friend unless the main tree has a TTreeIndex on the friend tree '%s'. You can also unset the "
                        "bit manually if you know what you are doing.";
```
Comment (Axel-Naumann) [inline tree/tree/src/TTree.cxx:1254]:
```suggestion
                       "bit manually if you know what you are doing; you risk associating wrong TTree entries of the friend wrt the main TTree!";
```
or similar would be even nicer :-)

**Review by Axel-Naumann — APPROVED** [review]

File: tree/tree/src/TTree.cxx (hunk 1 above)
Comment (eguiraud) [reply tree/tree/src/TTree.cxx]: done!

File: tree/tree/src/TTree.cxx [hunk 2]
```diff
@@ -1249,9 +1249,10 @@ bool CheckReshuffling(TTree &mainTree, TTree &friendTree)
 
    if ((isMainReshuffled || isFriendReshuffled) && !friendHasValidIndex) {
       const auto reshuffledTreeName = isMainReshuffled ? mainTree.GetName() : friendTree.GetName();
-      const auto msg = "Tree '%s' has the kEntriesReshuffled bit set, and cannot be used as friend nor can be added as "
-                       "a friend unless the main tree has a TTreeIndex on the friend tree '%s'. You can also unset the "
-                       "bit manually if you know what you are doing.";
+      const auto msg =
+         "Tree '%s' has the kEntriesReshuffled bit set and cannot have friends nor can be added as a friend unless the "
+         "main tree has a TTreeIndex on the friend tree '%s'. You can also unset the bit manually if you know what you "
```
Comment (pcanal) [inline tree/tree/src/TTree.cxx:1254]: I am not understanding the wording "the main tree has a TTreeIndex on the friend tree".   If I remember correctly in order for a friend to be properly read when "out of order", the index needs to be build in the friend. as the index contains the entry number corresponding to a given `Major/Minor' (eg Run Number /  Entry Number).   So we call `BuildIndex` on the friend (referring to column with the same names in both) and the main tree does not (directly) know about the index .

**Pull request merged**